### DISCUSSION TITLE
How to prevent search results from being displayed when the search term is empty

### DISCUSSION THREAD

**Discussion by ghost** [algolia/instantsearch#5294] [Q&A] [answered]
I am creating a full-text search function using react-instantsearch-dom.
How do I stop the search results from being displayed when the search term is empty?
I don't want to show the search results, but show "No search terms entered." I want to display "No search terms entered.

**Comment by francoischalifour** [comment, marked as answer]
We explain in our FAQ: [**How to search from the n-th character**](https://www.algolia.com/doc/guides/building-search-ui/troubleshooting/faq/react/#how-to-search-from-the-n-th-character).

Then you can use [`connectStateResults()`](https://www.algolia.com/doc/api-reference/widgets/state-results/react/) to display the placeholder text when there's no query.

**Reply by ghost** [reply]
thanks!!
